internal/distributed/datacoord/gen.go

```diff
@@ -0,0 +1,196 @@
+//go:build ignore
+// +build ignore
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"text/template"
+	"time"
+)
+
+//go:generate go run gen.go
+
+var methodTemplate = template.Must(template.New("").Parse(`// Code generated by go generate; DO NOT EDIT
+// This file is generated by go generate at {{ .Timestamp }}
+
+// Licensed to the LF AI & Data foundation under one
+// or more contributor license agreements. See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership. The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package grpcdatacoord
+
+import (
+	"context"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/milvus-io/milvus-proto/go-api/commonpb"
+	"github.com/milvus-io/milvus-proto/go-api/milvuspb"
+	"github.com/milvus-io/milvus/internal/proto/datapb"
+	"github.com/milvus-io/milvus/internal/proto/internalpb"
+	"github.com/milvus-io/milvus/internal/util/errorutil"
+	"github.com/milvus-io/milvus/internal/util/retry"
+)
+{{ range .Methods }}{{with .}}
+// {{.MethodComment}}
+func (s *Server) {{.MethodName}}(ctx context.Context, request {{.RequestType}}) ({{.ResponseType}}, error) {
+	var resp {{.ResponseType}}
+	var err error
+	retry.Do(ctx, func() error {
+		{{- if .SkipRequestParam}}
+		resp, err = s.dataCoord.{{.MethodName}}(ctx)
+		{{- else}}
+		resp, err = s.dataCoord.{{.MethodName}}(ctx, request)
+		{{- end}}
+		{{- if eq .ResponseType "*commonpb.Status"}}
+		if err == nil &&
+			resp.GetErrorCode() == commonpb.ErrorCode_UnexpectedError &&
+			strings.Contains(strings.ToLower(resp.GetReason()), errorutil.StandbyStateReason) {
+		{{- else}}
+		if err == nil &&
+			resp.GetStatus().GetErrorCode() == commonpb.ErrorCode_UnexpectedError &&
+			strings.Contains(strings.ToLower(resp.GetStatus().GetReason()), errorutil.StandbyStateReason) {
```

Comment:
We'd better use Error Code instead of matching reason string.

Reply:
good suggestion!